### PR TITLE
Update pagination to work with pgstac 0.9.X

### PR DESCRIPTION
**Pull request by zstatmanweil**
**Related Issue(s):**

- Addresses the pagination portion of #114

**Description:**

`pgstac` v0.9.x now returns prev and next tokens as links in the search returned from the database. This means the current version of the service with the most recent pgstac backend doesn't properly allow for pagination. The links returned by pgstac v0.9.x are formatted with a `GET` method request. At first I just passed in the search response's links as `extra_links` [here](https://github.com/stac-utils/stac-fastapi-pgstac/blob/main/stac_fastapi/pgstac/core.py#L271) in the code, which worked with GET requests but does not work with the formatting for POST requests. To simplify, I am now extracting the next and prev token from the links if they exist and don't change how they are handled in the rest of the code. 

I ensured tests passed with v0.8.5 and v0.9.1. I also ran GET and POST searches for both and confirmed the link responses were the same.
_v0.8.5 GET_
```"links":[
   {
      "rel":"next",
      "type":"application/geo+json",
      "method":"GET",
      "href":"http://localhost:8082/search?collection=joplin&token=next:joplin:d4eccfa2-7d77-4624-9e2a-3f59102285bb"
   },
   {
      "rel":"root",
      "type":"application/json",
      "href":"http://localhost:8082/"
   },
   {
      "rel":"self",
      "type":"application/json",
      "href":"http://localhost:8082/search?collection=joplin"
   }
]
```

_0.9.1 GET_
```"links":[
   {
      "rel":"root",
      "type":"application/json",
      "href":"http://localhost:8082/"
   },
   {
      "rel":"self",
      "type":"application/json",
      "href":"http://localhost:8082/search?collection=joplin"
   },
   {
      "rel":"next",
      "href":"http://localhost:8082/search?collection=joplin&token=next:joplin:d4eccfa2-7d77-4624-9e2a-3f59102285bb"
      "type":"application/geo+json",
      "method":"GET"
   }
]
```

_0.8.5 POST_
``` 
"links": [
        {
            "rel": "next",
            "type": "application/geo+json",
            "method": "POST",
            "href": "http://localhost:8082/search",
            "body": {
                "collections": [
                    "joplin"
                ],
                "sortby": [
                    {
                        "field": "properties.eo:cloud_cover",
                        "direction": "desc"
                    }
                ],
                "limit": 5,
                "token": "next:joplin:f2cca2a3-288b-4518-8a3e-a4492bb60b08"
            }
        },
        {
            "rel": "root",
            "type": "application/json",
            "href": "http://localhost:8082/"
        },
        {
            "rel": "self",
            "type": "application/json",
            "href": "http://localhost:8082/search"
        }
    ]
```

_0.9.0 POST_
``` 
"links": [
        {
            "rel": "next",
            "type": "application/geo+json",
            "method": "POST",
            "href": "http://localhost:8082/search",
            "body": {
                "collections": [
                    "joplin"
                ],
                "sortby": [
                    {
                        "field": "properties.eo:cloud_cover",
                        "direction": "desc"
                    }
                ],
                "limit": 5,
                "token": "next:joplin:f2cca2a3-288b-4518-8a3e-a4492bb60b08"
            }
        },
        {
            "rel": "root",
            "type": "application/json",
            "href": "http://localhost:8082/"
        },
        {
            "rel": "self",
            "type": "application/json",
            "href": "http://localhost:8082/search"
        }
    ]
```

**PR Checklist:**

- [X] `pre-commit` hooks pass locally
- [X] Tests pass (run `make test`)
- [X] Documentation has been updated to reflect changes, if applicable, and docs build successfully (run `make docs`)
- [X] Changes are added to the [CHANGELOG](https://github.com/stac-utils/stac-fastapi/blob/main/CHANGES.md).

### CODE REVIEW

**Comment by zstatmanweil** [comment]
I see the goal right now is probably to get 3.0.0 released so we can wait to merge this in until after that if folks like this change.

**Comment by vincentsarago** [comment]
thanks @zstatmanweil this looks great 🙏 

> I see the goal right now is probably to get 3.0.0 released so we can wait to merge this in until after that if folks like this change.

yeah, I've just pushed the release. I'll be back from holidays next week and I'll look at this PR 🙏

**Comment by zstatmanweil** [comment]
hey @vincentsarago ! Any thoughts on this PR?

**Comment by vincentsarago** [comment]
sorry @zstatmanweil I let this slipped 😓 

I think the PR looks good but we would need to add tests

**Comment by zstatmanweil** [comment]
No worries! I assumed, which is why I bugged. Ok sounds good, let me know if/how I can help.

File: setup.py
Comment (vincentsarago) [inline setup.py:20]: stac-fastapi-pgstac should be compatible with 0.8 and 0.9

File: setup.py
Comment (vincentsarago) [inline setup.py:26]: we use pgstac 0.9.x in tests

**Comment by vincentsarago** [comment]
@zstatmanweil I'll try to merge this week but we need to fix some issue with collection-search #155

**Comment by zstatmanweil** [comment]
Thanks for the update @vincentsarago !

**Comment by vincentsarago** [comment]
well it seems that we will get this PR merged first before #155

**Review by vincentsarago — APPROVED** [review]

**Comment by zstatmanweil** [comment]
@vincentsarago , do you know when you might cut a new release? I can always use "latest" if it isn't going to be soon.

**Comment by vincentsarago** [comment]
I'm waiting on a pastac release to finish some other PR 
sorry about the delay

**Pull request merged**
